code/items/equipments/entities/C4Entity.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Sandbox;
+using Sandbox.UI;
+
+using TTTReborn.Player;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_c4_ent"), Hammer.Skip]
+    public partial class C4Entity : Prop, IUse
+    {
+        private string ModelPath => "models/entities/c4.vmdl";
+
+        [Net]
+        public int AttachedBone { get; set; } = -1;
+
+        [Net]
+        public bool IsArmed { get; set; } = false;
+
+        //Timer display on C4 entity.
+        private WorldPanel Display;
+        private bool CreatedDisplay = false;
+
+        //Explosion parameters.
+        private const int RADIUS = 1024;
+        private const int DAMAGE = 500;
+        private const int FORCE = 50;
+
+        public override void Spawn()
+        {
+            base.Spawn();
+            SetModel(ModelPath);
+            SetupPhysicsFromModel(PhysicsMotionType.Dynamic);
+        }
+
+        public bool OnUse(Entity user)
+        {
+            TTTPlayer player = user as TTTPlayer;
+
+            //player.ClientOpenC4Control(State, this);
+
+            return false; //Only fire once
+        }
+
+        public bool IsUsable(Entity user)
+        {
+            if (user is TTTPlayer player)
+            {
+                return true;
+            }
+            return false;
```

Comment:
Does this work `return user is TTTPlayer`?

Reply:
Good call.